include/readout/models/IterableQueueModel.hpp

```diff
@@ -101,37 +111,12 @@ struct AccessableProducerConsumerQueue
     std::free(records_);
   }
 
-  template<class... Args>
-  bool write(Args&&... recordArgs)
-  {
-    // const std::lock_guard<std::mutex> lock(m_mutex);
-    auto const currentWrite = writeIndex_.load(std::memory_order_relaxed);
-    auto nextRecord = currentWrite + 1;
-    if (nextRecord == size_) {
-      nextRecord = 0;
-    }
-    // if (nextRecord == readIndex_.load(std::memory_order_acquire)) {
-    // std::cout << "SPSC WARNING -> Queue is full! WRITE PASSES READ!!! \n";
-    //}
-    //    new (&records_[currentWrite]) T(std::forward<Args>(recordArgs)...);
-    // writeIndex_.store(nextRecord, std::memory_order_release);
-    // return true;
-
-    // ORIGINAL:
-    if (nextRecord != readIndex_.load(std::memory_order_acquire)) {
-      new (&records_[currentWrite]) T(std::forward<Args>(recordArgs)...);
-      writeIndex_.store(nextRecord, std::memory_order_release);
-      return true;
-    }
-    // queue is full
+  bool put(T& record) { return write_(record); }
 
-    ++overflow_ctr;
-
-    return false;
-  }
+  bool write(T&& record) override { return write_(std::move(record)); }
```

Comment:
Eww. Why is this needed? I think the original write_ is anyway using placement new with perfect forwarding?